### PR TITLE
lint: get backend from format

### LINKED ISSUE #1963: Fix linter script issues
see https://github.com/mandiant/capa-rules/actions/runs/7725366127/job/21059427544

```
ERROR:lint:failed to extract capabilities: create directory /home/runner/work/capa-rules/capa-rules/tests/data/dotnet/692f7fd6d198e804d6af98eb9e390d61.exe_ unexpected format: dotnet
Traceback (most recent call last):
  File "/home/runner/work/capa-rules/capa-rules/scripts/lint.py", line 419, in check_rule
    capabilities = get_sample_capabilities(ctx, path)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/capa-rules/capa-rules/scripts/lint.py", line 368, in get_sample_capabilities
    extractor = capa.loader.get_extractor(
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/capa-rules/capa-rules/capa/loader.py", line 255, in get_extractor
    vw = get_workspace(input_path, input_format, sigpaths)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/capa-rules/capa-rules/capa/loader.py", line 157, in get_workspace
    raise ValueError("unexpected format: " + input_format)
ValueError: unexpected format: dotnet
```

### PR DESCRIPTION
**Pull request by mr-tz**
closes #1963

### Checklist

- [x] No CHANGELOG update needed

- [ ] No new tests needed

- [ ] No documentation update needed

### CODE REVIEW

File: scripts/lint.py [hunk 1]
```diff
@@ -364,11 +364,17 @@ def get_sample_capabilities(ctx: Context, path: Path) -> Set[str]:
...
+    fake_args = lambda: None
+    fake_args.backend = capa.main.BACKEND_AUTO
+    backend = capa.main.get_backend_from_cli(fake_args, format_)
+
     extractor = capa.loader.get_extractor(
         nice_path,
         format_,
         OS_AUTO,
-        capa.main.BACKEND_VIV,
+        backend,
```
Comment (williballenthin) [inline scripts/lint.py:377]: gross but ok.

would you add a one line explanation for why this is necessary? I think its: now that rules may reference non-samples, like CAPE results, then we want to be able to lint those examples, too.

**Review by williballenthin — APPROVED** [review]

File: scripts/lint.py [hunk 2]
```diff
@@ -356,19 +356,24 @@ def get_sample_capabilities(ctx: Context, path: Path) -> Set[str]:
...
+    format_ = capa.helpers.get_auto_format(nice_path)
+
+    # use format to determine one of the various backends (e.g. vivisect, dotnet, cape)
+    # create object to store backend attribute used by get_backend_from_cli
+    # TODO(mr-tz): refactor to avoid this hack
+    #  https://github.com/mandiant/capa/issues/1965
+    class FakeArgs:
+        backend = capa.main.BACKEND_AUTO
+
+    backend = capa.main.get_backend_from_cli(FakeArgs, format_)
```
Comment (williballenthin) [inline scripts/lint.py:370]:
```suggestion
```suggestion
    args = argparse.Namespace(format=capa.main.FORMAT_AUTO, backend=capa.main.BACKEND_AUTO)
    format_ = capa.main.get_input_format_from_args(args)
    backend = capa.main.get_backend_from_cli(args, format_)
```
```

Comment (mr-tz) [reply scripts/lint.py:370]: there we go :D

Comment (williballenthin) [reply scripts/lint.py:370]: TIL: about `argparse.Namespace` and how clean it is to use!

Comment (mr-tz) [reply scripts/lint.py:370]: should it be `get_input_format_from_cli`?

Comment (mr-tz) [reply scripts/lint.py]: ^^ changed

File: scripts/lint.py [hunk 3]
```diff
@@ -356,19 +356,21 @@ def get_sample_capabilities(ctx: Context, path: Path) -> Set[str]:
...
-    else:
-        format_ = capa.helpers.get_auto_format(nice_path)
-
     logger.debug("analyzing sample: %s", nice_path)
+
+    args = argparse.Namespace(
+        input_file=nice_path,
+        format=capa.main.FORMAT_AUTO,
+        backend=capa.main.BACKEND_AUTO
+    )
```
Comment (mr-tz) [inline scripts/lint.py:365]:
```suggestion
    args = argparse.Namespace(input_file=nice_path, format=capa.main.FORMAT_AUTO, backend=capa.main.BACKEND_AUTO)
```

**Review by williballenthin — APPROVED** [review]

**Pull request merged**